src/App.js

```diff
@@ -25,44 +24,97 @@ const generateSquares = () => {
   return squares;
 }
 
-const App = () => {
+const updateCurrentPlayer = (currentPlayer, setCurrentPlayer) => {
+  const updatePlayer = currentPlayer === PLAYER_1 ? PLAYER_2:PLAYER_1
+  setCurrentPlayer(updatePlayer)
+}
 
-  // This starts state off as a 2D array of JS objects with
-  // empty value and unique ids.
+const App = () => {
   const [squares, setSquares] = useState(generateSquares());
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1)
+  const [winner, setWinner ] = useState(null)
 
-  // Wave 2
-  // You will need to create a method to change the square 
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+  const checkForWinner = (squares) => {
+    let x = 0
+    let o = 0
+    let empty = false
+    const possibilities = [
+      [squares[0][0].value, squares[0][1].value, squares[0][2].value],
+      [squares[1][0].value, squares[1][1].value, squares[1][2].value],
+      [squares[2][0].value, squares[2][1].value, squares[2][2].value],
+      [squares[0][0].value, squares[1][0].value, squares[2][0].value],
+      [squares[0][1].value, squares[1][1].value, squares[2][1].value],
+      [squares[0][2].value, squares[1][2].value, squares[2][2].value],
+      [squares[0][0].value, squares[1][1].value, squares[2][2].value],
+      [squares[0][2].value, squares[1][1].value, squares[2][0].value]
+    ]
 
+    for(let i = 0; i < possibilities.length; i++){
+      if(possibilities[i][0] === '' || possibilities[i][1] === '' || possibilities[i][2] === ''){
+        empty = true
+      }
+      if(possibilities[i][0] === possibilities[i][1] && possibilities[i][1] === possibilities[i][2]){
+        if(possibilities[i][0] === 'x'){
+          x += 1
```

Comment:
We know that we're calling this function after every move, so as soon as we find three in a row (so long as it's not three blanks) we can stop looking. There should never be a case where we find more than 1 three in a row of xs and os.